graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -1160,14 +1160,41 @@ int64_t TemporalNumPick(
     const torch::optional<torch::Tensor>& edge_timestamp, int64_t seed_offset,
     int64_t offset, int64_t num_neighbors) {
   constexpr int64_t kFastPathThreshold = 1000;
-  if (num_neighbors > kFastPathThreshold && !probs_or_mask.has_value()) {
+  if (num_neighbors > kFastPathThreshold && !probs_or_mask.has_value() &&
+      fanout != -1) {
     // TODO: Currently we use the fast path both in TemporalNumPick and
     // TemporalPick. We may only sample once in TemporalNumPick and use the
     // sampled edges in TemporalPick to avoid sampling twice.
-    auto [success, sampled_edges] = FastTemporalPick(
-        seed_timestamp, csc_indics, fanout, replace, seed_pre_time_window,
-        node_timestamp, edge_timestamp, seed_offset, offset, num_neighbors);
-    if (success) return sampled_edges.size();
+    int64_t sampled_count = 0;
+    auto timestamp =
+        utils::GetValueByIndex<int64_t>(seed_timestamp, seed_offset);
+    for (int64_t edge_id = offset;
+         edge_id < offset + num_neighbors && sampled_count < fanout;
+         edge_id++) {
+      if (replace && sampled_count > 0) {
```

Comment:
could we move this check out from for loop to avoid unnecessary check for each iteration?

Reply:
This is required by the logic that this check should be done before each iteration.